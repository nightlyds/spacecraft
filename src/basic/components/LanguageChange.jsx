import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSortDown } from "@fortawesome/free-solid-svg-icons";

const LanguageChange = () => {
  let [language, takeLanguage] = useState(false);
  return (
    <div className="language-change">
      <div className="language-change-current-box">
        <span
          className="language-change-current"
          onClick={() => {
            takeLanguage(!language);
          }}
        >
          EN{" "}
          <FontAwesomeIcon
            icon={faSortDown}
            className="language-change-current-icon"
          />
        </span>
      </div>
      {language && (
        <div className="language-change-all">
          <span className="language-change-all-language language-change-all-language-first">EN</span>
          <span className="language-change-all-language">UK</span>
        </div>
      )}
    </div>
  );
};

export default LanguageChange;
