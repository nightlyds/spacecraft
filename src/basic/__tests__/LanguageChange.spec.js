import React from 'react';
import LanguageChange from '../components/LanguageChange.jsx';
import { shallow } from 'enzyme';

const setUp = (props) => shallow(<LanguageChange {...props} />);

describe("Language Change component tests", () => {
    let component;

    beforeEach(() => {
        component = setUp();
    })
    
    it("Snapshot test", () => {
        expect(component).toMatchSnapshot();
    })
})