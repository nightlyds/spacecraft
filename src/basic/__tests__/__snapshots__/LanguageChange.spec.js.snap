// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Change component tests Snapshot test 1`] = `
<div
  className="language-change"
>
  <div
    className="language-change-current-box"
  >
    <span
      className="language-change-current"
      onClick={[Function]}
    >
      EN
       
      <FontAwesomeIcon
        border={false}
        className="language-change-current-icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f0dd",
              "M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z",
            ],
            "iconName": "sort-down",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </div>
</div>
`;
